src/api/app/mixins/has_relationships.rb

```diff
@@ -81,9 +81,9 @@ def maintainers
     (direct_users + users_in_groups).uniq
   end
 
-  def reviewers
-    direct_users = relationships.with_users_and_roles_query.reviewers.pluck('users.login').map! { |user| User.find_by_login!(user) }
-    users_in_groups = relationships.with_groups_and_roles_query.reviewers.pluck('groups.title')
+  def maintainers_and_reviewers
```

Comment:
I think you can do all of this before instantiation with a scope and [.distinct](https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-distinct). Not that we'll have many instantiations to throw away but it might be more elegant and outside this weird mixin...